src/EventStore.Core/Services/Storage/IndexCommitterService.cs

```diff
@@ -440,6 +447,41 @@ public bool IsReplicated(int commitCount)
                 }
             }
         }
+
+        private Task _mergeIndexesTask;
+
+        public void Handle(ClientMessage.MergeIndexes message)
+        {
+            if (!IsAllowed(message.User, message.CorrelationId, message.Envelope)) return;
+            if (_mergeIndexesTask != null && _mergeIndexesTask.Status == TaskStatus.Running)
+            {
+                Log.Info("Index Merge Operation already running...");
+                MakeReplyForMergeIndexes(message);
+                return;
+            }
+            _mergeIndexesTask = _tableIndex.MergeIndexes().ContinueWith(t =>
+            {
+                _tableIndex.Scavenge(_logManager.CreateLog(), default(CancellationToken));
+            });
+            MakeReplyForMergeIndexes(message);
+        }
+
+        private static void MakeReplyForMergeIndexes(ClientMessage.MergeIndexes message)
+        {
+            message.Envelope.ReplyWith(new ClientMessage.MergeIndexesResponse(message.CorrelationId,
+                ClientMessage.MergeIndexesResponse.MergeIndexesResult.Started));
+        }
+
+        private bool IsAllowed(IPrincipal user, Guid correlationId, IEnvelope envelope)
+        {
+            if (user == null || (!user.IsInRole(SystemRoles.Admins) && !user.IsInRole(SystemRoles.Operations)))
+            {
+                envelope.ReplyWith(new ClientMessage.ScavengeDatabaseResponse(correlationId, ClientMessage.ScavengeDatabaseResponse.ScavengeResult.Unauthorized, null));
```

Comment:
@riccardone The reply message: `ClientMessage.ScavengeDatabaseResponse` seems incorrect. This `IsAllowed` check was also not present here previously.   Please don't change unrelated code between reviews since it's difficult to test and follow the changes.

Reply:
?? you asked to move logic to IndexCommitterService. Please don't ask to move logic is you don't want the correlated dependecies (libs, logc, methods, check) to be moved

Reply:
The IsAllowed method that was checking permissions is now removed. The permission check for the IndexMerge operation is now only performed in the http controller